src/bridge.rs

```diff
@@ -433,16 +461,50 @@ impl Bridge {
         })
     }
 
+    /// Returns a vector of all devices that are registered at this `Bridge`, sorted by their id's.
+    /// This function returns an error if `bridge.username` is `None`.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// # tokio_test::block_on(async {
+    /// let bridge = hueclient::Bridge::for_ip([192u8, 168, 0, 4])
+    ///    .with_user("rVV05G0i52vQMMLn6BK3dpr0F3uDiqtDjPLPK2uj");
+    /// for device in &bridge.get_all_devices().await.unwrap() {
+    ///     println!("{:?}", device);
+    /// }
+    /// # })
+    /// ```
+    pub async fn get_all_devices(&self) -> crate::Result<Vec<Device>> {
+        let url = format!("https://{}/clip/v2/resource/device", self.ip);
+        let resp: BridgeResponseV2<Device> = self.client.get(&url).send().await?.json().await?;
+        let mut devices = resp.get()?;
+        devices.sort_by(|a, b| a.id.cmp(&b.id));
+        Ok(devices)
+    }
+
+    pub async fn index_all_devices(&self) -> crate::Result<HashMap<String, Device>> {
+        let devices = self.get_all_devices().await?;
+        Ok(devices.into_iter().fold(
+            HashMap::new(),
+            |mut map: HashMap<String, Device>, device| {
+                map.insert(device.id.clone(), device);
+                map
+            },
+        ))
+    }
```

Comment:
I think the fold should be replaced by a `.map(|dev| (dev.id.clone(), dev)).collect()` here.

Reply:
Yep, I didn't think of this, easier to read!